packages/lesswrong/components/hooks/useDynamicTableOfContents.ts

```diff
@@ -24,15 +24,8 @@ export const useDynamicTableOfContents = ({
       return precalcuatedToc;
     }
 
-    if (!html) {
-      return {
-        html,
-        sections: []
-      }
-    }
-
     const { sections = [], html: tocHtml = null } =
-      extractTableOfContents(parseDocumentFromString(html)) ?? {};
+      extractTableOfContents(parseDocumentFromString(html ?? '')) ?? {};
```

Comment:
@jpaddison3 

> Something in this PR (I assume) is causing posts with no headers to display an empty ToC. Ex: https://forum.effectivealtruism.org/posts/iJSYZJJrLMigJsBeK/lizka-s-shortform

This is the fix for that. It turned out to be a pretty narrow edge case when there is no html content (true for Quick take posts). You might have seen it on some crossposts as well (like [this one](https://forum.effectivealtruism.org/posts/FBLkSfL96cDtf78ea/trying-to-do-more-good)), this is due to LW having a different `MIN_HEADINGS_FOR_TOC` to us which means a toc is sent even for posts with no heading, this one I'm not planning to fix.